Automattic/wp-calypso
# Replace withActiveTheme usage with withIsFSEActive to checking if theme is FSE enabled

#### Proposed Changes

* Replaced the usage of `useBlockEditorSettingsQuery` data query with the new `useActiveTheme` data query introduced in #64497. This gives us the benefit of fetching up-to-date theme data directly from the `block-templates` property, instead of waiting for a data sync on Jetpack sites.
* Deleted the `use-fse-status.ts` file it is not being used nor referenced anymore.
* Removed `is_fse_active` from Typescript interfaces.

~~There is still an issue with the "Edit"/"Customize" button on the themes page (mentioned here: https://github.com/Automattic/wp-calypso/issues/62755#issuecomment-1191412581). These changes to the /theme page-related component will not fix the issue yet.~~ This has been fixed by adding the `themeSlug` to the useActiveTheme data query.



#### Testing Instructions



* **Create a new blog or select an existing blog that still has a visible site setup list on the `/home` page**
* **Switch to a non-FSE theme (Dara):**

Navigate to the `/plans` page and scroll down to the "Advanced customization
" section. The button text should display "Start customizing" and navigate to the customizer page.

<img width="472" alt="image" src="https://user-images.githubusercontent.com/10482592/180313778-b9f499e7-f801-4d90-a884-634bcceb460f.png">

Go to the `/home` page and click the "Update your homepage" site setup task. The title for the card on the left side should display "Update your homepage".

<img width="951" alt="image" src="https://user-images.githubusercontent.com/10482592/180312564-3021d3d9-4ff5-432a-b5d4-d585d9c5570b.png">

Go to the `/themes` page and the "Customize" button should display inside the current theme card information at the top of the page. It should navigate to the customize page as well.
   
<img width="948" alt="image" src="https://user-images.githubusercontent.com/10482592/180314268-d927c58c-45b2-4226-a62f-fa1e45934e1d.png">

   
* **Switch to an FSE theme (Blank Canvas)**

(Must use blog with premium plan)
Navigate to the `/plans` page , select the `My plan` tab, and scroll down to the "Advanced customization
  " section. The button text should display "Edit site" and navigate to the site-editor page.

Go to the `/home` page and click the "Update your homepage" site setup task. The title for the card on the left side should display "Update your site's design".
   
Go to the `/themes` page and the "Edit" button should display inside the current theme card information at the top of the page. The button should navigate to the site-editor page as well.
   

#### Pre-merge Checklist






Related to #65697
